clarity/src/vm/analysis/type_checker/tests/contracts.rs

```diff
@@ -630,3 +664,1864 @@ fn test_expects() {
+        ))
+        (define-trait trait-b (
+            (echo (<trait-2>) (response uint uint))
+        ))
+        (define-public (wrapped-echo (contract <trait-a>))
+            (internal-echo contract))
+        (define-public (internal-echo (contract <trait-b>) (callee <trait-2>))
+            (contract-call? contract echo callee))";
+
+        let trait_with_superset_trait_arg = "(define-trait trait-1 (
+            (echo (uint) (response uint uint))
+        ))
+        (define-trait trait-2 (
+            (echo (uint) (response uint uint))
+            (foo (uint) (response uint uint))
+        ))
+        (define-trait trait-a (
+            (echo (<trait-1>) (response uint uint))
+        ))
+        (define-trait trait-b (
+            (echo (<trait-2>) (response uint uint))
+        ))
+        (define-public (wrapped-echo (contract <trait-a>))
+            (internal-echo contract))
+        (define-public (internal-echo (contract <trait-b>) (callee <trait-2>))
+            (contract-call? contract echo callee))";
+
+        let trait_with_subset_trait_arg = "(define-trait trait-1 (
+            (echo (uint) (response uint uint))
+        ))
+        (define-trait trait-2 (
+            (echo (uint) (response uint uint))
+            (foo (uint) (response uint uint))
+        ))
+        (define-trait trait-a (
+            (echo (<trait-1>) (response uint uint))
+        ))
+        (define-trait trait-b (
+            (echo (<trait-2>) (response uint uint))
+        ))
+        (define-public (wrapped-echo (contract <trait-b>))
+            (internal-echo contract))
+        (define-public (internal-echo (contract <trait-a>) (callee <trait-1>))
+            (contract-call? contract echo callee))";
+
+        let trait_with_duplicate_method = "(define-trait double-method (
+            (foo (uint) (response uint uint))
+            (foo (bool) (response bool bool))
+          ))";
+
+        let trait_to_subtrait_and_back = "(define-trait trait-1 (
+            (echo (uint) (response uint uint))
+        ))
+        (define-trait trait-2 (
+            (echo (uint) (response uint uint))
+            (foo (uint) (response uint uint))
+        ))
+        (define-private (foo-0 (impl-contract <trait-2>))
+            (foo-1 impl-contract))
+     
+        (define-private (foo-1 (impl-contract <trait-1>))
+            (foo-2 impl-contract))
+        
+        (define-private (foo-2 (impl-contract <trait-2>))
+            true)";
+
+        let trait_list_to_map = "(define-trait token-trait (
+            (echo (uint) (response uint uint))
+        ))
+        (define-public (send-many (data (list 10 {amount: uint, sender: principal, recipient: principal})) (token <token-trait>))
+            (ok (map my-iter data (list token token token token token token token token token token)))
+        )
+        (define-private (my-iter (data {amount: uint, sender: principal, recipient: principal}) (token <token-trait>))
+            true
```

Comment:
This test would be more convincing if it called some method in `token` on `data`.

Reply:
```
        (contract-call? token echo u5)
```
added in d247e4e4b17cbc4c0f43df638cf2af6a3d31ff94